W-12557250-sftp-release-2.0-kt (#2237)
mulesoft/docs-connectors: sftp/2.0/modules/ROOT/pages/sftp-connector-upgrade-guide.adoc

= Upgrade and Migrate SFTP Connector to Version 2.0  - Mule 4

Upgrade Anypoint Connector for SFTP (SFTP Connector) to version 2.0.0.

== Supported Upgrade Paths

[%header,cols="50a,50a"]
|===
|From Version | To Version
|1.6.0 |2.0.0
|===

== Changes in This Release

This release contains the following changes:

* The system property `mule.sftp.useHomeDirectory` was removed. SFTP Connector now uses the default home directory.
* SFTP Connector no longer supports the `jcraft library`. The connector now uses `apache mina library`.

> * Changed the *List* operation name to *List and get*. 

Reviewer: Is this just the name? or the operation xml is also changed?

= Upgrade and Migrate SFTP Connector to Version 2.0  - Mule 4

Upgrade Anypoint Connector for SFTP (SFTP Connector) to version 2.0.0.

== Supported Upgrade Paths

[%header,cols="50a,50a"]
|===
|From Version | To Version
|1.6.0 |2.0.0
|===

== Changes in This Release

This release contains the following changes:

* The system property `mule.sftp.useHomeDirectory` was removed. SFTP Connector now uses the default home directory.

> * SFTP Connector 2.0.0 no longer supports the `jcraft library`. The connector now uses `apache mina sshd library`.

Reviewer 2: ```suggestion
* SFTP Connector 2.0.0 no longer supports the `jcraft library`. The connector now uses the `apache mina sshd library`.
```

> * The *List* operation does not fetch the content of each file, instead, the payload of each list item contains the complete path to the found file or directory. 

Reviewer 2: ```suggestion
* The *List* operation no longer fetches the content of each file. Now, the payload of each list item contains the complete path to the found file or directory. 
```

* The SFTP *Regular Size* file attribute is renamed to *Regular File* because its behavior is not related to size. The file attribute value can be a file (`true`) or a directory (`false`).

> * SFTP Connector 2.0.0 does not support ciphers `3des-ctr` and `arcfour`. See <<supported-ciphers,Supported Ciphers project>>.

Reviewer 2: ```suggestion
* SFTP Connector 2.0.0 no longer supports ciphers `3des-ctr` and `arcfour`. For more information, refer to <<supported-ciphers,Supported Ciphers>>.
```

* The SFTP proxy protocol works only with HTTP and SOCKS5.

> * SFTP Connector no longer uses `javax.activation` to resolve the media types of the files. The connector now utilizes `Apache Tika`, which can result in different media file interpretations.

Reviewer 2: ```suggestion
* SFTP Connector no longer uses `javax.activation` to resolve the media types of the files. The connector now uses `Apache Tika`, which can result in different media file interpretations.
```
